config/base/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+//! Package for managing iroha configuration
+
+use serde::{de::DeserializeOwned, Serialize};
+use serde_json::Value;
+
+pub mod derive {
+    //! Modules for `Configurable` entities
+
+    use std::{error::Error as StdError, fmt};
+
+    use derive_more::Display;
+    /// Generate view for the type and implement conversion `Type -> View`.
+    /// View contains a subset of the fields that the type has.
+    ///
+    /// Works only with structs.
+    /// Type must implement `Default`.
+    ///
+    /// ## Container attributes
+    ///
+    /// ## Field attributes
+    /// ### `#[view(ignore)]`
+    /// Marks fields to ignore when converting to view type.
+    ///
+    /// ### `#[view(into = Ty)]`
+    /// Sets view's field type to Ty.
+    ///
+    /// ## Examples
+    ///
+    /// ```rust
+    /// use iroha_config_base::derive::view;
+    ///
+    /// view! {
+    ///     #[derive(Default)]
+    ///     struct Structure {
+    ///         #[view(type = u64)]
+    ///         a: u32,
+    ///         // `View` doesn't have field `b` so we must exclude it.
+    ///         #[view(ignore)]
+    ///         b: u32,
+    ///     }
+    /// }
+    ///
+    /// // Will generate something like
+    /// // --//-- original struct
+    /// //  struct StructureView {
+    /// //      a: u64,
+    /// //  }
+    /// //
+    /// //  impl From<Structure> for StructureView {
+    /// //      fn from(value: Structure) -> Self {
+    /// //          let Structure {
+    /// //              a,
+    /// //              ..
+    /// //          } = value;
+    /// //          Self {
+    /// //              a: From::<_>::from(a),
+    /// //          }
+    /// //      }
+    /// // }
```

Comment:
As I understood there will be also a `Default` impl, doesn't it?

Reply:
You are right, i will add it to the docs.